core/src/main/java/org/apache/iceberg/BaseTransaction.java

```diff
@@ -679,6 +680,10 @@ public LocationProvider locationProvider() {
     public String toString() {
       return name();
     }
+
+    Object writeReplace() {
```

Comment:
It is not clear to me from the name of this method what it does, I assume the return type is Object so we don't have to parameterize on the subtype?

Reply:
I am not sure but I thought Java serialization requires this signature.

Reply:
```Serializable classes that need to designate an alternative object to be used when writing an object to the stream should implement this special method with the exact signature:

 ANY-ACCESS-MODIFIER Object writeReplace() throws ObjectStreamException;
 
This writeReplace method is invoked by serialization if the method exists and it would be accessible from a method defined within the class of the object being serialized. Thus, the method can have private, protected and package-private access. Subclass access to this method follows java accessibility rules.

Classes that need to designate a replacement when an instance of it is read from the stream should implement this special method with the exact signature.

 ANY-ACCESS-MODIFIER Object readResolve() throws ObjectStreamException;```
 ? Should we throw objectStream Exception?

Reply:
https://docs.oracle.com/javase/8/docs/api/java/io/Serializable.html
I am learning things.

Reply:
Yep yep.